go/vt/orchestrator/logic/tablet_discovery.go

```diff
@@ -157,6 +158,31 @@ func refreshTabletsInCell(ctx context.Context, cell string, loader func(instance
 	refreshTablets(tablets, query, args, loader, forceRefresh)
 }
 
+// forceRefreshAllTabletsInShard is used to refresh all the tablet's information (both MySQL information and topo records)
+// for a given shard. This function is meant to be called before or after a cluster-wide operation that we know will
+// change the replication information for the entire cluster drastically enough to warrant a full forceful refresh
+func forceRefreshAllTabletsInShard(ctx context.Context, keyspace, shard string) {
+	log.Infof("force refresh of all tablets in shard - %v/%v", keyspace, shard)
+	if ctx == context.Background() {
+		refreshCtx, refreshCancel := context.WithTimeout(context.Background(), *topo.RemoteOperationTimeout)
+		defer refreshCancel()
+		ctx = refreshCtx
```

Comment:
I don't know if it is good practice to override the passed-in context like this..

Reply:
why do we need to check for the background context at all? we can apply `RemoteOperationTimeout` regardless. If the passed-in context has a deadline that is closer than current time + `RemoteOperationTimeout` that will anyway take precedence.

Reply:
I like this suggestion. I'll change the code.

Reply:
Fixed